src/core/prometheus.cc

```diff
@@ -800,19 +800,19 @@ class metrics_handler : public httpd::handler_base  {
 
     future<std::unique_ptr<http::reply>> handle(const sstring& path,
         std::unique_ptr<http::request> req, std::unique_ptr<http::reply> rep) override {
-        auto text = is_accept_text(req->get_header("Accept"));
+        auto text_format = !_ctx.allow_protobuf || !is_accept_protobuf(req->get_header("Accept"));
         sstring metric_family_name = req->get_query_param("__name__");
         bool prefix = trim_asterisk(metric_family_name);
         bool show_help = req->get_query_param("__help__") != "false";
         std::function<bool(const mi::labels_type&)> filter = make_filter(*req);
 
-        rep->write_body((text || !_ctx.allow_protobuf) ? "txt" : "proto", [this, text, metric_family_name, prefix, show_help, filter] (output_stream<char>&& s) {
+        rep->write_body((text_format) ? "txt" : "proto", [this, text_format, metric_family_name, prefix, show_help, filter] (output_stream<char>&& s) {
```

Comment:
could take this opportunity to drop the parenthesis around `text_format`. again, the variable name could be confusing. as it does not indicate the `text_format`.

Reply:
But It does, text_format means we are using text format

Reply:
sorry for the confusion. i mean, it does not contain the information about the "text_format", but about if the "text" content type is used. that's why i suggest to use name like "accepts_text" or "is_text_accepted" in previous comment.